# Changed - 2019-10-28 - jiapingzeng@ucla.edu
  timespec_cmp no longer returns false result when comparing far dates

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec-patched.h	2019-10-30 21:25:03.348725000 -0700
@@ -45,8 +45,13 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec == b.tv_sec) {
+  // time is same in seconds, compare nanoseconds
+    if (a.tv_nsec == b.tv_nsec) return 0;
+    return a.tv_nsec > b.tv_nsec ? 1 : -1;
+  }
+  // return 1 if a is greater, -1 if b is greater
+  return a.tv_sec > b.tv_sec ? 1 : -1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
